test/e2e/suite/conformance/certificatesigningrequests/venafi/cloud.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package venafi
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/jetstack/cert-manager/pkg/controller/certificatesigningrequests/util"
+	"github.com/jetstack/cert-manager/test/e2e/framework"
+	"github.com/jetstack/cert-manager/test/e2e/framework/addon/venafi"
+	"github.com/jetstack/cert-manager/test/e2e/framework/helper/featureset"
+	"github.com/jetstack/cert-manager/test/e2e/framework/util/errors"
+	"github.com/jetstack/cert-manager/test/e2e/suite/conformance/certificatesigningrequests"
+)
+
+var _ = framework.ConformanceDescribe("CertificateSigningRequests", func() {
+	// unsupportedFeatures is a list of features that are not supported by the
+	// Venafi TPP issuer.
+	var unsupportedFeatures = featureset.NewFeatureSet(
+		// Venafi TPP doesn't allow setting a duration
+		featureset.DurationFeature,
+		// Due to the current configuration of the test environment, it does not
+		// support signing certificates that pair with an elliptic curve or
+		// Ed255119 private keys
+		featureset.ECDSAFeature,
+		featureset.Ed25519FeatureSet,
+		// Our Venafi TPP doesn't allow setting non DNS SANs
+		// TODO: investigate options to enable these
+		featureset.EmailSANsFeature,
+		featureset.URISANsFeature,
+		featureset.IPAddressFeature,
```

Comment:
Can we either open an issue on GitHub to track this TODO and link it here, or transform the TODO into something else? eg.
```suggestion
		// Our Venafi TPP doesn't allow setting non DNS SANs because we don't
		// know how to configure our testing TPP instance to accept them.
		featureset.EmailSANsFeature,
		featureset.URISANsFeature,
		featureset.IPAddressFeature,
```